src/ruby/lib/ifsc.rb

```diff
@@ -0,0 +1,102 @@
+require 'json'
+require 'open-uri'
+
+module Razorpay
+  module IFSC
+    class InvalidCodeError < StandardError; end
+    class ServerError < StandardError; end
+
+    class Code
+      API = 'https://ifsc.razorpay.com/'
+
+      attr_reader :bank, :ifsc, :branch, :address, :contact, :city, :district, :state
+
+      def initialize(ifsc)
+        @ifsc = ifsc
+      end
+
+      def valid?
+        @valid ||= self.class.valid? ifsc
+      end
+
+      def get
+        raise InvalidCodeError unless valid?
+        @bank = api_data['BANK']
+        # @ifsc = data['ifsc']
+        @branch = api_data['BRANCH']
+        @address = api_data['ADDRESS']
+        @contact = api_data['CONTACT']
+        @city = api_data['CITY']
+        @district = api_data['DISTRICT']
+        @state = api_data['STATE']
+        self
+      end
+
+      private
+
+      def api_data
+        @api_data ||= JSON.parse(URI.join(API, ifsc).read)
+      rescue OpenURI::HTTPError, SocketError => e
+        raise ServerError, e.message
+      end
+
+      class << self
+        def find(ifsc)
+          self.new(ifsc).get
+        end
+
+        def validate!(code)
+          if valid? code
+            true
+          else
+            raise InvalidCodeError
+          end
+        end
+
+        def valid?(code)
+          return false unless code.size == 11
+          return false unless code[4] == '0'
+
+          bank_code = code[0..3].upcase
+          branch_code = code[5..-1].upcase
+
+          return false unless data.has_key? bank_code
+
+          list = data[bank_code]
+
+          if (branch_code.match(/^(\d)+$/))
+            lookup_numeric(list, branch_code)
+          else
+            lookup_string(list, branch_code)
+          end
+        end
+
+        private
+
+        def data
+          @data ||= JSON.load(File.read(File.join(__dir__, '../../IFSC.json')))
+        end
+
+        def lookup_numeric(list, branch_code)
+          branch_code = branch_code.to_i
+
+          return true if list.include? branch_code
+
+          lookup_ranges list, branch_code
```

Comment:
lookup_ranges is no longer supported because the underlying IFSC.json now holds everything